<center> <img src = https://raw.githubusercontent.com/AndreyRysistov/DatasetsForPandas/main/hh%20label.jpg alt="drawing" style="width:400px;">

# <center> Проект: Анализ вакансий из HeadHunter
   

Установка дополнительного пакета psycopg2

In [2]:
pip install psycopg2

Defaulting to user installation because normal site-packages is not writeable
   ---------------------------------------- 0.0/2.6 MB ? eta -:--:--
   ---- ----------------------------------- 0.3/2.6 MB ? eta -:--:--
   -------------------- ------------------- 1.3/2.6 MB 4.2 MB/s eta 0:00:01
   ---------------------------------------- 2.6/2.6 MB 5.1 MB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.2 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [6]:
import pandas as pd
import psycopg2

In [7]:
import warnings 
warnings.filterwarnings('ignore') 

In [36]:
connection = psycopg2.connect(
    dbname=DBNAME,
    user=USER,
    host=HOST,
    password=PASSWORD,
    port=PORT
)

NameError: name 'DBNAME' is not defined

<img src=https://lms.skillfactory.ru/asset-v1:SkillFactory+DST-3.0+28FEB2021+type@asset+block@SQL_pj2_2_1.png>

# Юнит 3. Предварительный анализ данных

1. Напишите запрос, который посчитает количество вакансий в нашей базе (вакансии находятся в таблице vacancies). 

In [9]:
# текст запроса
query_3_1 = f'''
    SELECT
      COUNT(id)
    FROM vacancies 
'''

In [11]:
# результат запроса
df = pd.read_sql_query(query_3_1 , connection)
display(df)

NameError: name 'connection' is not defined

2. Напишите запрос, который посчитает количество работодателей (таблица employers). 

In [12]:
# текст запроса
query_3_2 = f'''
    SELECT
      COUNT(id)
    FROM employers 
'''

In [13]:
# результат запроса
df = pd.read_sql_query(query_3_2 , connection)
display(df)

NameError: name 'connection' is not defined

3. Посчитате с помощью запроса количество регионов (таблица areas).

In [14]:
# текст запроса
query_3_3 = f'''
    SELECT 
      COUNT(id)
    FROM areas
'''

In [15]:
# результат запроса
df = pd.read_sql_query(query_3_3 , connection)
display(df)

NameError: name 'connection' is not defined

4. Посчитате с помощью запроса количество сфер деятельности в базе (таблица industries).

In [16]:
# текст запроса
query_3_4 = f'''
    SELECT 
      COUNT(id)
    FROM industries
'''

In [12]:
# результат запроса
df = pd.read_sql_query(query_3_4 , connection)
display(df)

count
0    294

***

### *Bыводы по предварительному анализу данных:* 
Ознакомилась с таблицами, их взаимосвязями и содержанием. В этом проекте мы будем работать с таблицами vacancies, employers, areas, industries.

# Юнит 4. Детальный анализ вакансий

1. Напишите запрос, который позволит узнать, сколько (cnt) вакансий в каждом регионе (area).
Отсортируйте по количеству вакансий в порядке убывания.

In [17]:
# текст запроса
query_4_1 = f'''
    SELECT 
      a.name,
      COUNT(v.name) AS cnt
    FROM vacancies v
      JOIN areas a ON v.area_id = a.id
    GROUP BY a.name
    ORDER BY COUNT(v.name) DESC 
    LIMIT 5 
'''

In [18]:
# результат запроса
df = pd.read_sql_query(query_4_1, connection)
display(df)

NameError: name 'connection' is not defined

2. Напишите запрос, чтобы определить у какого количества вакансий заполнено хотя бы одно из двух полей с зарплатой.

In [19]:
# текст запроса
query_4_2 = f'''
    SELECT 
      COUNT(*)
    FROM vacancies 
    WHERE (salary_from is not NULL) OR (salary_to is not NULL)
    
'''

In [20]:
# результат запроса
df = pd.read_sql_query(query_4_2, connection)
display(df)

NameError: name 'connection' is not defined

3. Найдите средние значения для нижней и верхней границы зарплатной вилки. Округлите значения до целого.

In [21]:
# текст запроса
query_4_3 = f'''
    SELECT 
      ROUND(AVG(salary_from),0)  AS "Нижняя граница",
      ROUND(AVG(salary_to),0) AS "Bерхняя граница"
    FROM vacancies
'''

In [22]:
# результат запроса
df = pd.read_sql_query(query_4_3, connection)
display(df)

NameError: name 'connection' is not defined

4. Напишите запрос, который выведет количество вакансий для каждого сочетания типа рабочего графика (schedule) и типа трудоустройства (employment), используемого в вакансиях. Результат отсортируйте по убыванию количества.


In [23]:
# текст запроса
query_4_4 = f'''
    SELECT 
      schedule,
      employment,
      COUNT(*)
    FROM vacancies 
    GROUP BY schedule, employment
    ORDER BY COUNT(*) DESC
'''

In [24]:
# результат запроса
df = pd.read_sql_query(query_4_4, connection)
display(df)

NameError: name 'connection' is not defined

5. Напишите запрос, выводящий значения поля Требуемый опыт работы (experience) в порядке возрастания количества вакансий, в которых указан данный вариант опыта. 

In [25]:
# текст запроса
query_4_5 = f'''
    SELECT 
      experience,
      COUNT(*)
    FROM vacancies 
    GROUP BY 1
    ORDER BY 2
'''

In [26]:
# результат запроса
df = pd.read_sql_query(query_4_5, connection)
display(df)

NameError: name 'connection' is not defined

***

### *Bыводы по детальному анализу вакансий*: 
Самый богатый на вакансии регион - это, конечно же, город Москва, следующий по списку город - Санкт-Петербург, и далее города-миллионники. Почти у половины вакансий не указана зароботная плата (всего - 49197, заполнено - 24073), это может означать, что  многие компании не ставят фиксированную з/п, т.к. она может зависеть от разных факторов и навыков сотрудника. Уровень зарплаты в среднем варьируется от 70 000 руб. до 110 000 руб. График работы по большинству вакансий предполагает полный рабочий день и полную занятость, в 5 раз меньше встречаются объявления с удаленным графиком. Можно сделать вывод, что офисные работники гораздо предпочтительнее, чем "удаленщики". Также можно отметить, что работодатели заинтересованы в соискателях с небольшим опытом работы от 1 до 3 лет, что предполагает молодым специалистам больше вариантов поиска.

# Юнит 5. Анализ работодателей

1. Напишите запрос, который позволит узнать, какие работодатели находятся на первом и пятом месте по количеству вакансий.

In [27]:
# текст запроса
query_5_1 = f'''
      SELECT 
        e.name AS "Работодатель",
        COUNT(v.id)
    FROM vacancies v
      JOIN employers e ON  e.id = v.employer_id
    GROUP BY 1
    ORDER BY 2 DESC
    LIMIT 5
'''

In [28]:
# результат запроса
df = pd.read_sql_query(query_5_1, connection)
display(df)

NameError: name 'connection' is not defined

2. Напишите запрос, который для каждого региона выведет количество работодателей и вакансий в нём.
Среди регионов, в которых нет вакансий, найдите тот, в котором наибольшее количество работодателей.


In [29]:
# текст запроса
query_5_2 = f'''
    SELECT 
      a.name as "Название ",
      COUNT(distinct e.id) AS "Количество работодателей",
      COUNT(distinct v.id) AS "Количество вакансий"
    FROM areas a 
      LEFT JOIN employers e ON a.id = e.area 
      LEFT JOIN vacancies v ON v.area_id = a.id
    WHERE v.id IS NULL
    GROUP BY a.name
    ORDER BY COUNT(distinct e.id) DESC
    LIMIT 3
    
'''

In [30]:
# результат запроса
df = pd.read_sql_query(query_5_2, connection)
display(df)

NameError: name 'connection' is not defined

3. Для каждого работодателя посчитайте количество регионов, в которых он публикует свои вакансии. Отсортируйте результат по убыванию количества.


In [31]:
# текст запроса
query_5_3 = f'''
    SELECT 
        e.name AS "Работодатель",
        COUNT(DISTINCT v.area_id) AS "Количество регионов"    
    FROM employers e
      LEFT JOIN vacancies v ON e.id = v.employer_id 
    GROUP BY 1
    ORDER BY 2 DESC
    LIMIT 10
'''

In [32]:
# результат запроса
df = pd.read_sql_query(query_5_3, connection)
display(df)

NameError: name 'connection' is not defined

4. Напишите запрос для подсчёта количества работодателей, у которых не указана сфера деятельности. 

In [29]:
# текст запроса
query_5_4 = f'''
    SELECT 
        COUNT(e.name)
    FROM employers e
      LEFT JOIN employers_industries ei ON e.id = ei.employer_id
      LEFT JOIN industries i ON ei.industry_id = i.id
    WHERE i.name IS NULL
'''

In [33]:
# результат запроса
df = pd.read_sql_query(query_5_4, connection)
display(df)

NameError: name 'query_5_4' is not defined

5. Напишите запрос, чтобы узнать название компании, находящейся на третьем месте в алфавитном списке (по названию) компаний, у которых указано четыре сферы деятельности. 

In [34]:
# текст запроса
query_5_5 = f'''
      SELECT 
        e.name,
        COUNT(i.id)
    FROM employers e
      LEFT JOIN employers_industries ei ON e.id = ei.employer_id
      LEFT JOIN industries i ON ei.industry_id = i.id
    GROUP BY e.name
    HAVING COUNT(i.id)=4
    ORDER BY e.name
    LIMIT 1 OFFSET 2
'''

In [35]:
# результат запроса
df = pd.read_sql_query(query_5_5, connection)
display(df)

NameError: name 'connection' is not defined

6. С помощью запроса выясните, у какого количества работодателей в качестве сферы деятельности указана Разработка программного обеспечения.


In [33]:
# текст запроса
query_5_6 = f'''
      SELECT 
        i.name,
        COUNT(e.id)
    FROM employers e
      LEFT JOIN employers_industries ei ON e.id = ei.employer_id
      LEFT JOIN industries i ON ei.industry_id = i.id
    WHERE i.name = 'Разработка программного обеспечения'
    GROUP BY 1
    ORDER BY 2 DESC
'''

In [34]:
# результат запроса
df = pd.read_sql_query(query_5_6, connection)
display(df)

name  count
0  Разработка программного обеспечения   3553

7. Для компании «Яндекс» выведите список регионов-миллионников, в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавьте строку Total с общим количеством вакансий компании. Результат отсортируйте по возрастанию количества.

Список городов-милионников надо взять [отсюда](https://ru.wikipedia.org/wiki/%D0%93%D0%BE%D1%80%D0%BE%D0%B4%D0%B0-%D0%BC%D0%B8%D0%BB%D0%BB%D0%B8%D0%BE%D0%BD%D0%B5%D1%80%D1%8B_%D0%A0%D0%BE%D1%81%D1%81%D0%B8%D0%B8). 

Если возникнут трудности с этим задание посмотрите материалы модуля  PYTHON-17. Как получать данные из веб-источников и API. 

In [35]:
# код для получения списка городов-милионников

import urllib.parse as req

path_common = req.urlparse('https://ru.wikipedia.org/wiki/Города-миллионеры_России')
url = path_common.scheme + '://' + req.quote(path_common.netloc) +  req.quote(path_common.path) + '#'+ req.quote(path_common.fragment)
cities = pd.read_html(url)[1]
cities_millioner = tuple(cities['Город'].tolist())

In [36]:
# текст запроса
query_5_7 = f'''
    (SELECT 
      a.name AS "Название города",
      COUNT(v.id) AS "Количество вакансий"
    
    FROM areas a
      LEFT JOIN vacancies v ON a.id = v.area_id
      LEFT JOIN employers e ON v.employer_id = e.id
    GROUP BY a.name, e.name
    HAVING a.name in {cities_millioner} AND e.name = 'Яндекс')
    
    UNION ALL

    SELECT 'Total', SUM(x.cnt)
    FROM 
      (SELECT 
        COUNT(*) AS cnt
      FROM areas a
        LEFT JOIN vacancies v ON a.id = v.area_id
        LEFT JOIN employers e ON v.employer_id = e.id
      GROUP BY a.name, e.name
      HAVING a.name in {cities_millioner} AND e.name = 'Яндекс') as x
    ORDER BY 2 ASC  
'''

In [37]:
# результат запроса
df = pd.read_sql_query(query_5_7, connection)
display(df)

Название города  Количество вакансий
0              Омск                 21.0
1         Челябинск                 22.0
2        Красноярск                 23.0
3         Волгоград                 24.0
4             Пермь                 25.0
5            Казань                 25.0
6    Ростов-на-Дону                 25.0
7            Самара                 26.0
8               Уфа                 26.0
9         Краснодар                 30.0
10          Воронеж                 32.0
11      Новосибирск                 35.0
12  Нижний Новгород                 36.0
13     Екатеринбург                 39.0
14  Санкт-Петербург                 42.0
15           Москва                 54.0
16            Total                485.0

***

### *Bыводы по анализу работодателей:*
В ТОП-5 компаний, предлагающих большое количество вакансий входят: "Яндекс" - 1993,"Ростелеком" - 491,
"Тинькофф" - 444,"СБЕР" - 428,"Газпром нефть" - 331. Также из анализа видно, что компании "Яндекс" и "Ростелеком" публикуют большое количесво вакансий в разных регионах России.
У 8419 компаний сфера деятельности не указана, и соискатели не видят эти вакансии только потому, что системе поиска не хватает дополнительной информации для ранжирования.

# Юнит 6. Предметный анализ

1. Сколько вакансий имеет отношение к данным?

Считаем, что вакансия имеет отношение к данным, если в её названии содержатся слова 'data' или 'данн'.

*Подсказка: Обратите внимание, что названия вакансий могут быть написаны в любом регистре.* 


In [38]:
# текст запроса
query_6_1 = f'''
    SELECT 
      COUNT(name)
    FROM vacancies 
    WHERE LOWER(name) LIKE '%data%'
      OR LOWER(name) LIKE '%данн%'
'''

In [39]:
# результат запроса
df = pd.read_sql_query(query_6_1, connection)
display(df)

count
0   1771

2. Сколько есть подходящих вакансий для начинающего дата-сайентиста? 
Будем считать вакансиями для дата-сайентистов такие, в названии которых есть хотя бы одно из следующих сочетаний:
* 'data scientist'
* 'data science'
* 'исследователь данных'
* 'ML' (здесь не нужно брать вакансии по HTML)
* 'machine learning'
* 'машинн%обучен%'

** В следующих заданиях мы продолжим работать с вакансиями по этому условию.*

Считаем вакансиями для специалистов уровня Junior следующие:
* в названии есть слово 'junior' *или*
* требуемый опыт — Нет опыта *или*
* тип трудоустройства — Стажировка.
 

In [40]:
# текст запроса
query_6_2 = f'''
    SELECT
      COUNT(*)
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')
          AND ((name) ILIKE '%junior%'
            OR (experience) ILIKE '%Нет опыта%'
            OR (employment) ILIKE '%Стажировка%')
'''

In [41]:
# результат запроса
df = pd.read_sql_query(query_6_2, connection)
display(df)

count
0     51

3. Сколько есть вакансий для DS, в которых в качестве ключевого навыка указан SQL или postgres?

** Критерии для отнесения вакансии к DS указаны в предыдущем задании.*

In [42]:
# текст запроса
query_6_3 = f'''
    SELECT
      COUNT(*)
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')
          AND ((key_skills) ILIKE '%SQL%'
           OR (key_skills) ILIKE '%postgres%')     
'''

In [43]:
# результат запроса
df = pd.read_sql_query(query_6_3, connection)
display(df)

count
0    201

4. Проверьте, насколько популярен Python в требованиях работодателей к DS.Для этого вычислите количество вакансий, в которых в качестве ключевого навыка указан Python.

** Это можно сделать помощью запроса, аналогичного предыдущему.*

In [44]:
# текст запроса
query_6_4 = f'''
    SELECT 
      COUNT(*)
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')
          AND ((key_skills) ILIKE '%Python%')
'''

In [45]:
# результат запроса
df = pd.read_sql_query(query_6_4, connection)
display(df)

count
0    351

5. Сколько ключевых навыков в среднем указывают в вакансиях для DS?
Ответ округлите до двух знаков после точки-разделителя.

In [46]:
# текст запроса
query_6_5 = f'''
    SELECT 
      ROUND(AVG((LENGTH(key_skills) - LENGTH(replace(key_skills,CHR(9),'')))::integer+1), 2) as "Среднее количество"
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')
'''

In [47]:
# результат запроса
df = pd.read_sql_query(query_6_5, connection)
display(df)

Среднее количество
0                6.41

6. Напишите запрос, позволяющий вычислить, какую зарплату для DS в **среднем** указывают для каждого типа требуемого опыта (уникальное значение из поля *experience*). 

При решении задачи примите во внимание следующее:
1. Рассматриваем только вакансии, у которых заполнено хотя бы одно из двух полей с зарплатой.
2. Если заполнены оба поля с зарплатой, то считаем зарплату по каждой вакансии как сумму двух полей, делённую на 2. Если заполнено только одно из полей, то его и считаем зарплатой по вакансии.
3. Если в расчётах участвует null, в результате он тоже даст null (посмотрите, что возвращает запрос select 1 + null). Чтобы избежать этой ситуацию, мы воспользуемся функцией [coalesce](https://postgrespro.ru/docs/postgresql/9.5/functions-conditional#functions-coalesce-nvl-ifnull), которая заменит null на значение, которое мы передадим. Например, посмотрите, что возвращает запрос `select 1 + coalesce(null, 0)`

Выясните, на какую зарплату в среднем может рассчитывать дата-сайентист с опытом работы от 3 до 6 лет. Результат округлите до целого числа. 

In [48]:
# текст запроса
query_6_6 = f'''
    SELECT 
        experience AS "Опыт работы",
        ROUND(AVG(coalesce((salary_from+salary_to)/2, salary_from, salary_to)), 0) AS "Средняя зарплата"
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')
    GROUP BY experience   
'''

In [49]:
# результат запроса
df = pd.read_sql_query(query_6_6, connection)
display(df)

Опыт работы  Средняя зарплата
0         Более 6 лет               NaN
1           Нет опыта           74643.0
2  От 1 года до 3 лет          139675.0
3       От 3 до 6 лет          243115.0

***

### *Bыводы по предметному анализу:*

Это была самая полезная часть проекта. В ней мы узнали о делах на рынке труда для специалиста по Data Sсience. Вакансий для начинающего специалиста не так уж и много (~51), но если владеть некоторыми навыками, например, SQL или postgres, то предложения вырастают в 4 раза. В среднем рабодотодатель указывает около 6 ключевых навыков в своих вакансиях. В большом количестве вакансий указано требования о знании языка Python.

# Общий вывод по проекту

## <center> Дополнительное исследование

* Общее кол-во вакансий DS

In [50]:
# текст запроса
query_1 = f'''
    SELECT
        COUNT(*) 
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')   
'''

In [51]:
# результат запроса
df = pd.read_sql_query(query_1, connection)
display(df)

count
0    480

* Разброс вакансий  DS по регионам

In [52]:
# текст запроса
query_2 = f'''
    SELECT 
      a.name AS "Город",
      COUNT(v.name) AS cnt
    FROM vacancies v
      JOIN areas a ON v.area_id = a.id
    WHERE 
        ((v.name) ILIKE '%data scientist%'
        OR (v.name) ILIKE '%data science%'
        OR (v.name) ILIKE '%исследователь данных%'
        OR ((v.name) LIKE '%ML%' AND (v.name) NOT ILIKE '%HTML%')
        OR (v.name) ILIKE '%machine learning%'
        OR (v.name) ILIKE '%машинн%обучен%') 
    GROUP BY a.name
    ORDER BY 2 DESC 
    LIMIT 5 
'''

In [53]:
# результат запроса
df = pd.read_sql_query(query_2, connection)
display(df)

Город  cnt
0           Москва  217
1  Санкт-Петербург   64
2      Новосибирск   23
3  Нижний Новгород   20
4           Казань   16

* Количество вакансий для топ-3 основных типов рабочего графика и типа трудоустройства, с указание средней зароботной платы

In [54]:
# текст запроса
query_3 = f'''
    SELECT 
      schedule,
      employment,
      COUNT(*),
      ROUND(AVG(coalesce((salary_from+salary_to)/2, salary_from, salary_to)), 0) AS "Средняя зарплата"
    FROM vacancies
    WHERE 
        ((name) ILIKE '%data scientist%'
        OR (name) ILIKE '%data science%'
        OR (name) ILIKE '%исследователь данных%'
        OR ((name) LIKE '%ML%' AND (name) NOT ILIKE '%HTML%')
        OR (name) ILIKE '%machine learning%'
        OR (name) ILIKE '%машинн%обучен%')  
    GROUP BY 1,2
    ORDER BY 3 DESC
    LIMIT 3
'''

In [55]:
# результат запроса
df = pd.read_sql_query(query_3, connection)
display(df)

schedule        employment  count  Средняя зарплата
0       Полный день  Полная занятость    319          178307.0
1  Удаленная работа  Полная занятость    110          209409.0
2     Гибкий график  Полная занятость     37          175000.0

In [56]:
connection.close()

### *Обобщеные выводы по вакансии DS:*

Профессия дата сайентиста востребована. Но да, большинство вакансий сосредоточены в Москве и Санкт-Петербурге. На hh.ru сейчас примерно 480 вакансий по запросу Data Scientist. При этом больше 217 из них приходятся на столицу, 64 — на Питер, немногим больше 20 — на пару городов-миллиоников, а дальше числа стремительно уменьшаются. Но переезжать не обязательно. Можно работать на удаленке, несмотря на то, что преждложений по удаленной работе меньше в 3 раза, чем работы в офисе, средняя з/п выше примерно на 20%.

## <center> Подведем итог исследования

Давайте сделаем общий вывод:
- Больше всего вакансий в Москве(5333) и Санкт-Петербурге(2851). 
- Средние значения нижней и верхней зарплатной вилки ~71065 и ~110537. 
- Большинство вакансий у нас на полный рабочий день с полной занятостью, потом численность вакансий резко падает.  
- Среди регионов, в которых нет вакансий, в котором наибольшее количество работодателей является Россия(410). 
- Самой большой компанией является Яндекс, она публикует вакансии в 181 регионе. Oбщее количество вакансий компании Яндекс для городов-миллионников 485, самым большим среди вакансий является Москва- 54.
- Количество работодателей у которых не указана сфера деятельности - 8419. 
- У 3553 вакансий работодателей указана "Разработка программного обеспечения". 
- Мы также определили, что кандидат с опытом работы от 3 до 6 лет может претендовать на заработную плату в размере ~243115 руб. Количество вакансий уменьшается по мере роста опыта.
- В наших данных есть 1771 которые имеют отношения к данным. Для начинающих дата-сайентистов доступно только 51 из них, что является довольно небольшим количеством. Однако, если использовать SQL или Postgres в качестве ключевого навыка, то количество вакансий почти увеличится в 4 раза. Стоит учесть, что включение этих навыков может привести к тому, что поиск уже не будет ограничиваться только начинающими специалистами. Количество вакансий, где требуется использование языка Python, составляет 351. В среднем для вакансий, связанных с Data Science, требуется 6,41 ключевых навыков. 